### ISSUE TITLE
Ksgs LES turbulence model wall BC

### ISSUE THREAD

**Issue by msv5450** [NaluCFD/Nalu#424]
The transport equation for the Ksgs LES turbulence model doesn't allow a no flux BC on a wall boundary. Does it have any physical bearing that it must have a Dirichlet BC for turbulent kinetic energy on the wall BCs?  

I made a slight change in the source code and it runs fine with a no flux BC.

**Comment by spdomin** [comment]
For a fully resolved wall boundary condition, I believe that the zero flux weak bc is suitable for Ksgs. As the mesh is refined, the wall value should approach zero.

For a non-resolved wall simulation, i.e., using wall functions, one must modify the production and dissipation source terms assembled to the near wall nodal value to be in local equilibrium. If one performs this assumption, then in the limit of small advection and diffusion, the wall value of Ksgs should approach (utau^2)/sqrt(Cmu) - exactly what the wall-function approach prescribes.

See: https://nalu.readthedocs.io/en/latest/source/theory/boundaryConditions.html

Bottom line is that for a resolved flow, zero flux would be an alternative to the Dirichlet condition of zero. For a wall function, I think that you need zero flux in concert with modification of the near wall production term. 

Nalu has the wall function value written up under: https://nalu.readthedocs.io/en/latest/source/theory/boundaryConditions.html

You can probably find the SNL/Fuego theory manual online (open access) and read more about alternative approaches to Dirichlet approaches for wall-modeled approach.

**Comment by spdomin** [comment]
Answered.
